src/pooled.rs

```diff
@@ -1,10 +1,108 @@
-//! A connection pool to a SQL database.
+//! # A connection pool to a SQL database.
+//!
+//! A pool is created through the [`builder`] method, starting from a connection
+//! string that allows some of the parameters be delivered by the user.
+//!
+//! A connection string has the following structure:
+//!
+//! `connector_type://user:password@host/database?parameters`
+//!
+//! Connector type can be one of the following:
+//!
+//! - `sqlite`/`file` opens an SQLite connection
+//! - `mysql` opens a MySQL connection
+//! - `postgres`/`postgresql` opens a PostgreSQL connection
+//!
+//! All parameters should be given in the query string format:
+//! `?key1=val1&key2=val2`. All parameters are optional.
+//!
+//! ## Common parameters
+//!
+//! - `connection_limit` defines the number of connections opened to the
+//!   database.
+//!
+//! ## SQLite
+//!
+//! - `user`/`password` do not do anything and can be emitted.
+//! - `host` should point to the database file.
```

Comment:
do we really parse it as a URL on sqlite? As far as I remember it was something we stopped doing because it didn't work nicely with some characters, including spaces, in file paths. Should we mention that in the docs?

Reply:
We don't use `Url` but we treat the string as one.